packages/preferences/src/browser/util/preference-tree-generator.ts

```diff
@@ -69,8 +69,8 @@ export class PreferenceTreeGenerator {
     });
 
     protected createLeafNode = (property: string, preferencesGroup: Preference.Branch): SelectableTreeNode => {
-        const propertySpecifier = this.split(property).slice(1);
-        const name = propertySpecifier.map(word => word.slice(0, 1).toLocaleUpperCase() + word.slice(1)).join(' ').trim();
+        const rawLeaf = property.split('.').pop();
+        const name = this.formatString(rawLeaf!);
```

Comment:
Looks good to me. @vince-fugnitto, I'm a bit wary  of the non-null assertion, but I've seen it elsewhere around the repo, and this seems like a fairly safe instance to me. What are your thoughts?

An option to avoid it would be to have `.formatString` return an empty string when passed `undefined`.

Reply:
We could also do the following @colin-grant-work?

```ts
const rawLeaf: string = property.split('.').pop() || property;
const name = this.formatString(rawLeaf);
```

It would be a safeguard against preferences which do not follow the format of being `a.b` but rather `a`.

Reply:
I think it's functionally equivalent to the current code, but it would make Typescript happier 😄 . If there's no `.`, then the `split` will just return an array with the whole string at index 0.

Reply:
> I think it's functionally equivalent to the current code, but it would make Typescript happier  . If there's no `.`, then the `split` will just return an array with the whole string at index 0.

Correct :) if you are wary about uses of it we can think about adding an `eslint` rule against it: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-non-null-assertion.md. I believe it would benefit the project to be more strict regarding `null`-checking.